File: backend/src/main/java/app/netbooks/backend/models/Subscription.java
package app.netbooks.backend.models;

import java.sql.Date;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class Subscription {
    private Long id;
    private UUID subscriber;
    private Integer edition;
    private Date startedIn;
    private Date closedIn;
    private Boolean automaticBilling;
    private Boolean actived;

    public Subscription(UUID subscriber, Integer edition) {
        this.subscriber = subscriber;
        this.edition = edition;
    };
};

File: frontend/src/models/subscription.ts
import { UUID } from "crypto";
import { PlanEdition, PlanEditionData } from "./plan_edition";

export type SubscriptionData = {
  id: number;
  subscriber: UUID;
  edition: PlanEditionData;
  startedIn: Date;
  closedIn?: Date;
  dueDate: Date;
  payDate: Date;
  numPayments: number;
  actived: boolean;
};

export class Subscription {
  readonly id: number;
  readonly subscriber: UUID;
  readonly edition: PlanEdition;
  readonly startedIn: Date;
  readonly closedIn?: Date;
  readonly dueDate: Date;
  readonly payDate: Date;
  readonly numPayments: number;
  readonly actived: boolean;

  constructor(data: SubscriptionData) {
    this.id = data.id;
    this.subscriber = data.subscriber;
    this.edition = new PlanEdition(data.edition);
    this.startedIn = new Date(data.startedIn);
    this.closedIn = data.closedIn ? new Date(data.closedIn) : undefined;
    this.dueDate = new Date(data.dueDate);
    this.payDate = new Date(data.payDate);
    this.numPayments = data.numPayments;
    this.actived = data.actived;
  }
}
